python/multicorn/sqlalchemyfdw.py

```diff
@@ -440,20 +478,21 @@ def _build_statement(self, quals, columns, sortkeys):
             statement = statement.order_by(column)
         return statement
 
-    def execute(self, quals, columns, sortkeys=None):
+    def execute(self, quals, columns, sortkeys=None, aggs=None, group_clauses=None):
         """
         The quals are turned into an and'ed where clause.
         """
         sortkeys = sortkeys or []
-        statement = self._build_statement(quals, columns, sortkeys)
+        is_aggregation = aggs or group_clauses
+        statement = self._build_statement(quals, columns, sortkeys, aggs=aggs, group_clauses=group_clauses)
         log_to_postgres(str(statement), DEBUG)
 
         # If a dialect doesn't support streaming using server-side cursors,
         # we hack it a bit by sending LIMIT/OFFSET queries.
         offset = 0
         with inject_envvars(self.envvars):
             while True:
-                if self.batch_size is not None:
+                if self.batch_size is not None and not is_aggregation:
```

Comment:
Do we really need to disable pagination when aggregating (i.e. does `SELECT agg(x), y FROM z GROUP BY y LIMIT 10 OFFSET 20` not work properly with aggregations)? If you have a test with a small `batch_size` (smaller than the aggregation's return size), does it break with this change removed?

Reply:
No need really, I just left it like that to simplify debugging edge cases and forgot about it.

Fwiw, I did test it with batching turned on, and a small batch size option, and it was correct, bringing it back now.